### PR TITLE
No updates if current version is newer than announced version.

### PR DESCRIPTION
**Pull request by KodeCR**
Current versions of deCONZ are much newer than the announced version and since it was a simple compare not equal it always showed an update was available.
Changed it so only if a newer version is announced an update is shown.

### CODE REVIEW

File: discovery.cpp [hunk 1]
```diff
@@ -359,13 +359,21 @@ void DeRestPluginPrivate::internetDiscoveryExtractVersionInfo(QNetworkReply *rep
 
             if (!version.isEmpty())
             {
-                if (gwUpdateVersion != version)
+                QStringList gwUpdateVersionList = gwUpdateVersion.split('.');
+                QStringList versionList = version.split('.');
+                if (((gwUpdateVersionList[0].toInt() <  versionList[0].toInt())) ||
```
Comment (manup) [inline discovery.cpp:364]: I'd recommend to add a check that both lists contain at least 3 values:

    (gwUpdateVersionList.size() >= 3 && versionList.size() >= 3)

**Review by manup — CHANGES_REQUESTED** [review]

**Comment by manup** [comment]
Hi, good catch, can you please add the check to prevent potential SEGV.

File: discovery.cpp (hunk 1 above)
Comment (KodeCR) [reply discovery.cpp]: Thanks, done

**Comment by KodeCR** [comment]
Thanks, just noticed this now and have added the check.

**Pull request merged**